homeassistant/components/xiaomi_miio/switch.py

```diff
@@ -113,120 +87,111 @@
 }
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the switch from config."""
-    if DATA_KEY not in hass.data:
-        hass.data[DATA_KEY] = {}
-
-    host = config[CONF_HOST]
-    token = config[CONF_TOKEN]
-    name = config[CONF_NAME]
-    model = config.get(CONF_MODEL)
-
-    _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
-
-    devices = []
-    unique_id = None
-
-    if model is None:
-        try:
-            miio_device = Device(host, token)
-            device_info = await hass.async_add_executor_job(miio_device.info)
-            model = device_info.model
-            unique_id = f"{model}-{device_info.mac_address}"
-            _LOGGER.info(
-                "%s %s %s detected",
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the switch from a config entry."""
+    entities = []
+
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+        if DATA_KEY not in hass.data:
+            hass.data[DATA_KEY] = {}
+
+        host = config_entry.data[CONF_HOST]
+        token = config_entry.data[CONF_TOKEN]
+        name = config_entry.data[CONF_NAME]
+        model = config_entry.data[CONF_MODEL]
+        unique_id = config_entry.unique_id
+
+        _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
```

Comment:
Debug level max.

How long is the full token? Do we need to log this?

Reply:
Full token is 32 chars long, I just copied this, it was already there but happy to remove it, its not really nessesary.
For now I will change it to debug level, let me know if you want me to remove it.